src/sentry/auth/superuser.py

```diff
@@ -204,6 +204,10 @@ def is_privileged_request(self) -> tuple[bool, InactiveReason]:
         """
         Returns ``(bool is_privileged, RequestStatus reason)``
         """
+
+        # Dynamic import is used to avoid circular dependency issues
+        has_completed_sso = importlib.import_module("sentry.auth", "has_completed_sso")
+
```

Comment:
what was the cycle here?  this can be a normal inline import without importlib and it should work too

importlib means we lose typing

Reply:
Makes sense. superuser.py is imports sentry.auth, and sentry.auth imports superuser